app/controllers/stats_controller.rb

```diff
@@ -55,6 +55,15 @@ def index
     end
   end
 
+  def translation_report
+    @params = params
+
+    respond_to do |format|
+      format.html
+      format.csv { render text: generate_translation_words_csv }
+    end
```

Comment:
please split the GET of translation report HTML out from the POST for the generated CSV into two separate actions. No need to overload into 1.